docs/examples/ratify-verify-azure-cmd.md

```diff
@@ -1,296 +1,351 @@
+1. Configure AKV resource names.
 
-### Sign the image using ```notation```
+    ```bash
+    # Name of the existing Azure Key Vault used to store the signing keys
+    AKV_NAME=<your-unique-keyvault-name>
+    # New desired key name used to sign and verify
+    KEY_NAME=wabbit-networks-io
+    CERT_PATH=./${KEY_NAME}.pem
+    ```
 
-1. We will build, push, sign the image in ACR
+2. Configure ACR and image resource names.
 
     ```bash
-    # Build, push, sign the image in ACR
-    echo $IMAGE
+    # Name of the existing registry example: myregistry.azurecr.io
+    ACR_NAME=<your-registry-name>
+    # Existing full domain of the ACR
+    REGISTRY=$ACR_NAME.azurecr.io
+    # Repository name inside ACR where image will be stored
+    REPO=net-monitor
+    TAG=v1
+    IMAGE=$REGISTRY/${REPO}:$TAG
+    # Source code directory containing Dockerfile to build
+    IMAGE_SOURCE=https://github.com/wabbit-networks/net-monitor.git#main
     ```
 
-2. Build and push the image
+## Create and store the self-signed certificate in AKV
+
+If you have an existing certificate, you can upload it to AKV. For more information on how to use your own signing key, see the [signing certificate requirements.](https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#certificate-requirements).
+
+Otherwise create an x509 self-signed certificate storing it in AKV for remote signing using the steps below.
+
+1. Create a certificate policy file.
+
+    Once the certificate policy file is executed as below, it creates a valid signing certificate compatible with **notation** in AKV. The EKU listed is for code-signing, but isn't required for notation to sign artifacts.
 
     ```bash
-    # build the image
-    docker build -t $IMAGE https://github.com/wabbit-networks/net-monitor.git#main
+    cat <<EOF > ./my_policy.json
+    {
+        "issuerParameters": {
+        "certificateTransparency": null,
+        "name": "Self"
+        },
+        "x509CertificateProperties": {
+        "ekus": [
+        "1.3.6.1.5.5.7.3.3"
+        ],
+        "keyUsage": [
+          "digitalSignature"
+        ],
+        "subject": "CN=wabbit-networks.io,O=Notary,L=Seattle,ST=WA,C=US",
+        "validityInMonths": 12
+        }
+    }
+    EOF
+    ```
 
-    # push the image
-    docker push $IMAGE
+2. Create the certificate.
+
+    ```azure-cli
+    az keyvault certificate create -n $KEY_NAME --vault-name $AKV_NAME -p @my_policy.json
     ```
 
-3. Generate a test certificate
+3. Get the Key ID for the certificate.
+
+    ```bash
+    KEY_ID=$(az keyvault certificate show -n $KEY_NAME --vault-name $AKV_NAME --query 'kid' -o tsv)
+    ```
+4. Download public certificate.
 
     ```bash
-    # Generate a test certificate
-    notation cert generate-test --default "wabbit-networks.io"
+    CERT_ID=$(az keyvault certificate show -n $KEY_NAME --vault-name $AKV_NAME --query 'id' -o tsv)
+    az keyvault certificate download --file $CERT_PATH --id $CERT_ID --encoding PEM
```

Comment:
Not a blocker but curious why we need to download the public cert here?

Reply:
@akashsinghal  The downloaded cert is used to be added to the trust store for signature verification. 

I just removed the [duplicated image signing steps](https://learn.microsoft.com/en-us/azure/container-registry/container-registry-tutorial-sign-build-push)g from this doc and mentioned it is a prerequisite. This will make this doc not too long and keep single-source maintenance of content.

Reply:
Ahh that makes sense. Thanks for the explanation